motor-control/firmware/motor_control_hardware.c

```diff
@@ -0,0 +1,24 @@
+/*
+ * A C file to act as a middle layer to HAL related functions for GPIO.
+ */
+
+#include "motor_control_hardware.h"
+
+#include "FreeRTOS.h"
+#include "task.h"
+
+#include "platform_specific_hal_conf.h"
+
+void motor_hardware_set_pin(void* port, uint16_t pin, uint8_t active_setting) {
+    HAL_GPIO_WritePin(port, pin, active_setting);
+}
+
+void motor_hardware_reset_pin(void* port, uint16_t pin, uint8_t active_setting) {
+    HAL_GPIO_WritePin(
+        port, pin,
+        (active_setting == GPIO_PIN_SET) ? GPIO_PIN_RESET : GPIO_PIN_RESET);
```

Comment:
What is the purpose of this statement? Is this supposed to support active low pins? This looks wrong.

Reply:
yeah the idea is to support active-low pins. This is "turning the pin off" (i.e. set to low if active high, high if active low)

Reply:
could probably be a bit easier to read